pyfr/libs/conduit.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+
+from ctypes import (POINTER, c_char_p, c_void_p, 
+                    c_double, c_float, c_long, c_longlong, RTLD_GLOBAL)
+from pyfr.ctypesutil import LibWrapper
+from typing import Union
+
+
+class ConduitWrappers(LibWrapper):
+    _libname = 'conduit'
+
+    # Functions
+    _functions = [
+        (c_void_p, 'conduit_node_append', c_void_p),
+        (c_void_p, 'conduit_node_create', c_void_p),
+        (c_void_p, 'conduit_node_destroy', c_void_p),
+        (c_void_p, 'conduit_node_fetch', c_void_p, c_char_p),
+        (c_char_p, 'conduit_node_fetch_path_as_char8_str', c_void_p, 
+         c_char_p),
+        (POINTER(c_float), 'conduit_node_fetch_path_as_float32_ptr', c_void_p,
+         c_char_p),
+        (POINTER(c_double), 'conduit_node_fetch_path_as_float64_ptr', 
+         c_void_p, c_char_p),
+        (POINTER(c_longlong), 'conduit_node_fetch_path_as_int64_ptr', 
+         c_void_p, c_char_p),
+        (c_longlong, 'conduit_node_total_bytes_allocated', c_void_p),
+        (c_void_p, 'conduit_node_set_path_char8_str', c_void_p, c_char_p,
+         c_char_p),
+        (c_void_p, 'conduit_node_set_path_float32', c_void_p, c_char_p,
+         c_float),
+        (c_void_p, 'conduit_node_set_path_float32_ptr', c_void_p, c_char_p,
+         POINTER(c_double), c_longlong),
+        (c_void_p, 'conduit_node_set_path_float64', c_void_p, c_char_p, 
+         c_double),
+        (c_void_p, 'conduit_node_set_path_float64_ptr', c_void_p, c_char_p,
+         POINTER(c_double), c_longlong),
+        (c_void_p, 'conduit_node_set_path_int32', c_void_p, c_char_p, c_long),
+        (c_void_p, 'conduit_node_set_path_int64', c_void_p, c_char_p,
+         c_longlong),
+        (c_void_p, 'conduit_node_set_path_int64_ptr', c_void_p, c_char_p,
+         POINTER(c_longlong), c_longlong),
+        (c_void_p, 'conduit_node_set_path_node', c_void_p, c_char_p, c_void_p), # maybe this is copying, external might be better
+        (c_void_p, 'conduit_node_print', c_void_p),
+        (c_void_p, 'conduit_node_print_detailed', c_void_p),
+    ]
+
+    def __init__(self):
+        super().__init__(mode=RTLD_GLOBAL)
+
+
+class Conduit:
+    def __init__(self):
+        self.lib = ConduitWrappers()
+
+
+class ConduitNode:
+    def __init__(self, ptr=None):
+        cd = Conduit()
+        self.lib = cd.lib
+
+        if isinstance(ptr, c_void_p):
+            self.ptr = ptr
+        else:
+            self.ptr = c_void_p(self.lib.conduit_node_create(c_void_p()))
```

Comment:
No need for the cast as it already returns a `c_void_p` and can just pass in `None`.

Reply:
Passing none I agree with, but the c_void_p(...) is needed, otherwise it seg faults.

Reply:
The prototype is: `CONDUIT_API conduit_node *conduit_node_create();` which suggests it should not take any arguments.